storage-adapter/app/com/microsoft/azure/iotsolutions/storageadapter/services/DocDBKeyValueContainer.java

```diff
@@ -106,18 +102,22 @@ public void delete(String collectionId, String key) throws DocumentClientExcepti
         }
     }
 
-
-    public Status ping() throws CreateResourceException {
-        createDocumentClientLazily();
-        URI response = null;
-        if (this.client != null) {
-            response = this.client.getReadEndpoint();
-        }
-        if (response != null) {
-            return new Status(true, "Alive and Well!");
-        } else {
-            return new Status(false, "Could not connect to DocumentDb." + collectionLink);
+    public StatusResultServiceModel ping() {
+        StatusResultServiceModel result = new StatusResultServiceModel(false, "Storage check failed");
+        try {
+            createDocumentClientLazily();
```

Comment:
this.createDocumentClientLazily(). Also I know createDocumentClientLazily was already there, but private methods should be after public methods

Reply:
done